helper_apps/MediaFileChunker/fileChunker.js

```diff
@@ -117,31 +98,90 @@ function convertYoutubeToMp3Stream(video) {
 
 async function pipeStreamToFile(stream, filePath) {
     try {
-        await cPipeline(stream, fs.createWriteStream(filePath));
+        await pipeline(stream, fs.createWriteStream(filePath));
         console.log('Stream piped to file successfully.');
     } catch (error) {
         console.error(`Error piping stream to file: ${error.message}`);
     }
 }
 
+const saveYoutubeUrl = async (url, filename) => {
+    let stream = ytdl(url, {
+        quality: 'highestaudio',
+    });
 
-const processYoutubeUrl = async (url) => {
-    const info = await ytdl.getInfo(url);
-    const audioFormat = ytdl.chooseFormat(info.formats, { quality: 'highestaudio' });
+    return new Promise((resolve, reject) => {
+        ffmpeg(stream)
+            .audioBitrate(128)
+            .save(filename)
+            .on('progress', p => {
+                readline.cursorTo(process.stdout, 0);
+                process.stdout.write(`${p.targetSize}kb downloaded`);
+            })
+            .on('error', (err) => {
+                console.log('an error happened: ' + err.message);
+                reject(err);
+            })
+            .on('end', () => {
+                console.log(`\ndone, thanks - ${(Date.now() - start) / 1000}s`);
```

Comment:
Is `start` undef here?

Reply:
this is the older version not used anymore i'll get rid of it